### PR TITLE
feat: Add error library and examples for error management

### PR DESCRIPTION
**Pull request by fteychene**
### Why ths PR ?

This PR add `anyhow` and `thiserror` crates for clean error management.  
The PR come with examples of how you can use this crates for error declaration, context additon and downcasting for a good basic usage of these crates in [examples/errors](https://github.com/dhall-mock/dhall-mock/pull/10/files#diff-39fef3daa36097d29ccbf7d1102a3a0dR42)

To run error examples : `cargo run --package dhall-mock --example errors`

### Related
Closes: #4

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
 
-fn main() {
+fn main() -> Result<(), Error> {
     println!("Hello from dhall mock project 👋");
 
     // Some Dhall data
     let data = r###"
         let Mock = ./dhall/Mock/package.dhall
-        in Mock.HttpMethod.GET
+        in Mock.HttpMethod.UNKNOW
```
Comment (rlecomte) [inline src/main.rs:10]: could you revert this change to keep the main in success?

Comment (fteychene) [reply src/main.rs]: :+1:
